typescript/web/src/pages/[workspaceSlug]/datasets/[datasetSlug]/index.tsx

```diff
@@ -30,6 +30,10 @@ const Body = () => {
     skip: typeof datasetSlug !== "string" || typeof workspaceSlug !== "string",
   });
 
+  if (error) {
+    router.push("/404");
+  }
```

Comment:
Having an error with Apollo client does not necessarily means that the entity does not exists. We have to be more specific and keep the usual error handling when appropriate.

Reply:
What is the usual error handling ?

Reply:
Rely on `typescript/web/src/pages/_error.tsx` IIRC